services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/document/EnhancedDocument.java

```diff
@@ -0,0 +1,616 @@
+         * @param values        The ByteBuffer values that needs to be set.
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder addBinarySet(String attributeName, ByteBuffer... values);
+
+        /**
+         * Appends an attribute of name attributeName with specified list of values to the document builder.
+         *
+         * @param attributeName Name of the attribute that needs to be added in the Document.
+         * @param value         The list of values that needs to be set.
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder addList(String attributeName, List<?> value);
+
+        /**
+         * Appends an attribute of name attributeName with specified list of values to the document builder.
+         *
+         * @param attributeName Name of the attribute that needs to be added in the Document.
+         * @param values        The list of values that needs to be set.
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder addList(String attributeName, Object... values);
+
+        /**
+         * Appends an attribute of name attributeName with specified map values to the document builder.
+         *
+         * @param attributeName Name of the attribute that needs to be added in the Document.
+         * @param value         The map that needs to be set.
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder addMap(String attributeName, Map<String, ?> value);
+
+        /**
+         * Appends an attribute of name attributeName with specified value of the given JSON document in the form of a string.
+         *
+         * @param attributeName Name of the attribute that needs to be added in the Document.
+         * @param json          JSON document in the form of a string.
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder addJson(String attributeName, String json);
+
+        /**
+         * Convenience builder methods that sets the attributes of this documents from the specified KeyAttributeMetadata
+         * components.
+         *
+         * @param components KeyAttributeMetadata of the attributes
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder keyComponents(KeyAttributeMetadata... components);
+
+        /**
+         * Convenience builder methods that sets an attribute of this document for the specified key attribute name and value.
+         *
+         * @param keyAttrName  Name of the attribute that needs to be added in the Document.
+         * @param keyAttrValue The value that needs to be set.
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder keyComponent(KeyAttributeMetadata keyAttrName, Object keyAttrValue);
```

Comment:
What's the use-case for it?

Reply:
This is from V1 where user provide KeyAttributes directly . As of now will ad this in IM , will discuss this in API surface review